### PR TITLE
s390 part of JDK-8281146

### PR DESCRIPTION
**Pull request by RealLucy**
---------
### Progress
- [x] Change must not contain extraneous whitespace
- [ ] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Reviewers
 * [Martin Doerr](https://openjdk.java.net/census#mdoerr) (@TheRealMDoerr - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7604/head:pull/7604` \
`$ git checkout pull/7604`

Update a local copy of the PR: \
`$ git checkout pull/7604` \
`$ git pull https://git.openjdk.java.net/jdk pull/7604/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7604`

View PR using the GUI difftool: \
`$ git pr show -t 7604`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7604.diff">https://git.openjdk.java.net/jdk/pull/7604.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/cpu/s390/s390.ad [hunk 1]
```diff
@@ -10273,14 +10273,13 @@ instruct string_inflate_const(Universe dummy, iRegP src, iRegP dst, iRegI tmp, i
 %}
 
 // StringCoding.java intrinsics
-instruct has_negatives(rarg5RegP ary1, iRegI len, iRegI result, roddRegI oddReg, revenRegI evenReg, iRegI tmp, flagsReg cr) %{
-  match(Set result (HasNegatives ary1 len));
-  effect(TEMP_DEF result, USE_KILL ary1, TEMP oddReg, TEMP evenReg, TEMP tmp, KILL cr); // R0, R1 are killed, too.
+instruct count_positives(rarg5RegP ary1, iRegI len, iRegI result, iRegI tmp, flagsReg cr) %{
+  match(Set result (CountPositives ary1 len));
+  effect(TEMP_DEF result, USE_KILL ary1, TEMP tmp, KILL cr); // R0, R1 are killed, too.
```
Comment (TheRealMDoerr) [inline src/hotspot/cpu/s390/s390.ad:10278]: You preserve ary1 in the new implementation. So, you can use iRegP ary1 and remove the USE_KILL effect.

**Review by TheRealMDoerr — COMMENTED** [review]
Looks correct. I have a minor suggestion.

File: src/hotspot/cpu/s390/s390.ad (hunk 1 above)
Comment (RealLucy) [reply src/hotspot/cpu/s390/s390.ad:10278]: Can I be sure then that ary1 isn't allocated the same register as result. Does the effect TEMP_DEF result prevent that? I vaguely remember there was an issue with the initial implementation of has_negatives.

Comment (TheRealMDoerr) [reply src/hotspot/cpu/s390/s390.ad:10278]: Yes, TEMP(_DEF) and USE registers are disjoint.

**Review by TheRealMDoerr — APPROVED** [review]
Latest updates look good. I guess you need to ask Claes to integrate them, because he has already integrated the earlier commits into his PR.

**Comment by RealLucy** [comment]
> Latest updates look good. I guess you need to ask Claes to integrate them, because he has already integrated the earlier commits into his PR.

Thank you for the review, Martin. I have already informed @cl4es about the new, and essential, commits.

**Pull request closed without merging**